services/web/server/src/simcore_service_webserver/products/_middlewares.py

```diff
@@ -69,4 +69,8 @@ async def discover_product_middleware(request: web.Request, handler: Handler):
 
         request[RQ_PRODUCT_KEY] = product_name
 
+    assert (  # nosec
+        request.get(RQ_PRODUCT_KEY) is not None or request.path == "/socket.io/"
```

Comment:
this is something I wanted to ask already some time ago after discussions with MD, why is the path of socket.io missing the product name?

Reply:
I guess the reason is that we never used/needed that information in the sio handler
AFAIK the front-end only adds X-Product header in the http API calls but the soi client is a different one (and therefore controls the http calls himself)

Reply:
ok all clear. I thought the frontend was passing these cookies on every single request.